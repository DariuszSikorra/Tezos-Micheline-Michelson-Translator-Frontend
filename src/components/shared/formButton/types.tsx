export type ButtonProps = {
  type: 'submit' | 'reset' | 'button';
  label: string;
  className?: string;
  onClick?: any;
  disabled?: boolean;
  stylingType?: string;
};
