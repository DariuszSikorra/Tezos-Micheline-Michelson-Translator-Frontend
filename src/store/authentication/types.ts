export const AUTHENTICATION_SET_AUTH = 'AUTHENTICATION_SET_AUTH';
export const AUTHENTICATION_LOGIN = 'AUTHENTICATION_LOGIN';
export const AUTHENTICATION_SUCCESS = 'AUTHENTICATION_SUCCESS';
export const AUTHENTICATION_FAIL = 'AUTHENTICATION_FAIL';


export type authCredentials = {
  username: string,
  password: string,
};

export type AuthenticationState = {
  token: string;
  isLogged: string;
};

export interface IAuthenticationLogin {
  type: typeof AUTHENTICATION_LOGIN,
  payload: authCredentials,
};
export interface IAuthenticationSuccess {
  type: typeof AUTHENTICATION_SUCCESS,
  token: string,
};
export interface IAuthenticationFail {
  type: typeof AUTHENTICATION_FAIL,
  error: string,
  status: string
};

export type AuthActionTypes =
  | IAuthenticationLogin
  | IAuthenticationSuccess
  | IAuthenticationFail;