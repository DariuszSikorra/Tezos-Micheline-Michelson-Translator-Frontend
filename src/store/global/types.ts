import { LibraryState } from "../library/types";
import { TranslatorState, TranslatorMessageState } from "../translator/types";
import { MessageState } from "../message/types";
import { AuthenticationState } from "../authentication/types";

export interface IState {
  library: LibraryState;
  message: MessageState;
  auth: AuthenticationState;
  translator: TranslatorState;
  translatorMessage: TranslatorMessageState;
  pending: boolean;
  error?: string;
}