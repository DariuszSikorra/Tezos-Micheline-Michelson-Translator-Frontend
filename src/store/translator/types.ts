export const TRANSLATOR_SET_MODE = 'TRANSLATOR_SET_MODE';
export const TRANSLATOR_FETCH_MICHELSON_TO_MICHELINE = 'TRANSLATOR_FETCH_MICHELSON_TO_MICHELINE';
export const TRANSLATOR_FETCH_MICHELINE_TO_MICHELSON = 'TRANSLATOR_FETCH_MICHELINE_TO_MICHELSON';
export const TRANSLATOR_SET_MICHELINE = 'TRANSLATOR_SET_MICHELINE';
export const TRANSLATOR_SET_MICHELSON = 'TRANSLATOR_SET_MICHELSON';
export const TRANSLATOR_SET_ERROR = 'TRANSLATOR_SE_ERROR';
export const TRANSLATOR_FLUSH_TRANSLATION = 'TRANSLATOR_FLUSH_TRANSLATION';

export type TranslatorState = {
  mode: Modes;
  michelson: string;
  micheline: string;
  error: string;
};

export type TranslatorAction = {
  type: string;
  mode: Modes;
  translation: string;
  error?: any;
};

export enum Modes {
  MICHELINEMICHELSON = 'michelinemichelson',
  MICHELSONMICHELINE = 'michelsonmicheline',
}

export interface IState {
  pending: boolean;
  translator: TranslatorState;
  error?: any;
}

export interface ITranslatorSetMode {
  type: typeof TRANSLATOR_SET_MODE;
  mode: Modes;
}

export interface ITranslatorFetchMichelineToMichelson {
  type: typeof TRANSLATOR_FETCH_MICHELINE_TO_MICHELSON;
  payload: string;
}

export interface ITranslatorFetchMichelsonToMicheline {
  type: typeof TRANSLATOR_FETCH_MICHELSON_TO_MICHELINE;
  payload: string;
}

export interface ITranslatorSetMichelson {
  type: typeof TRANSLATOR_SET_MICHELSON;
  status: number;
  translation: string;
}

export interface ITranslatorSetMicheline {
  type: typeof TRANSLATOR_SET_MICHELINE;
  status: number;
  translation: string;
}

export interface ITranslatorSetError {
  type: typeof TRANSLATOR_SET_ERROR;
  error: string;
}

export interface ITranslatorFlushTranslation {
  type: typeof TRANSLATOR_FLUSH_TRANSLATION;
}

export type TranslatorActionTypes =
  | ITranslatorSetMode
  | ITranslatorFetchMichelineToMichelson
  | ITranslatorFetchMichelsonToMicheline
  | ITranslatorSetMicheline
  | ITranslatorSetMichelson
  | ITranslatorSetError
  | ITranslatorFlushTranslation;
