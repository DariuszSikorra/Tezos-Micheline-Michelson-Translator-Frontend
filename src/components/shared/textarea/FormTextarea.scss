.form-textarea {

   .form-textarea_input {
       resize: none;
       width: calc(100% - 30px);
       border: 1px solid #787878;
       border-radius: 3px;
       min-height: 50px;
       padding: 15px;
       font-size: 16px;
       line-height: 24px;
       color: rgba(0, 0, 0, 0.87);

       &:focus {
           border-color: #C91C00;
       }
   }

   .form-textarea_label {
    color: #C91C00;
    position: relative;
    background: white;
    width: 100px;
    top: 10px;
    left: 20px;
    text-align: center;
   }

   .form-textarea_error {
    height: 20px;
    text-align: left;
    opacity: 0.6;
    font-size: 12px;
   }
}