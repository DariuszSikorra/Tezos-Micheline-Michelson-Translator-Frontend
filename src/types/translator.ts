export type ModesType = 'michelinemichelson' | 'michelsonmicheline';

export const Labels = {
  michelinemichelson: 'Micheline to Michelson',
  michelsonmicheline: 'Michelson to Micheline'
};
