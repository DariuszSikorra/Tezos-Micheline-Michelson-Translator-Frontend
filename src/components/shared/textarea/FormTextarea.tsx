import React from 'react';
import './FormTextarea.scss';

const FormTextarea = ({
    label,
    placeholder,
    name,
    type,
    onChange,
    onBlur,
    value,
    errors,
    touched,
    className,
}: any) => (
        <div className={`form-textarea ${className}`}>
            <div className='form-textarea_label'>
                {label}
            </div>
            <textarea
                className='form-textarea_input'
                placeholder={placeholder}
                name={name}
                onChange={onChange}
                onBlur={onBlur}
                value={value}
                onClick={() => console.log(onBlur)}
            />
            <div className="form-textarea_error">
                {errors && touched && errors}
            </div>
        </div>
    );

export default FormTextarea;