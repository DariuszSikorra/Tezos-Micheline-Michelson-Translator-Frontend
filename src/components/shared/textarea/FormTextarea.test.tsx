import React from 'react';
import { mount } from 'enzyme';
import FormTextarea from './FormTextarea';
import { FormTextareaProps } from './types';

const defaultProps: FormTextareaProps = {
    label: 'label',
    name: 'name',
    placeholder: 'placeholder',
    onChange: () => { },
    onBlur: () => { },
    value: 'test value',
    errors: undefined,
    touched: false,
    className: '',
};

describe("FormTextarea", () => {
    const wrapper = mount(<FormTextarea {...defaultProps} />);

    it("matches snapshot", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("renders without crashing", () => {
        const item = wrapper.find(FormTextarea);

        expect(
            item
                .at(0)
                .find(".form-textarea_input")
                .at(0)
                .text()
        ).toEqual('test value')
    });
});