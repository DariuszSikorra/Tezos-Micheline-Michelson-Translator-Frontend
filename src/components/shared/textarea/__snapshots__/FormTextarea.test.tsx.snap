// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextarea matches snapshot 1`] = `
"<FormTextarea label=\\"label\\" name=\\"name\\" placeholder=\\"placeholder\\" onChange={[Function: onChange]} onBlur={[Function: onBlur]} value=\\"test value\\" errors={[undefined]} touched={false} className=\\"\\">
  <div className=\\"form-textarea \\">
    <div className=\\"form-textarea_label\\">
      label
    </div>
    <textarea className=\\"form-textarea_input\\" placeholder=\\"placeholder\\" name=\\"name\\" onChange={[Function: onChange]} onBlur={[Function: onBlur]} value=\\"test value\\" />
    <div className=\\"form-textarea_error\\" />
  </div>
</FormTextarea>"
`;
